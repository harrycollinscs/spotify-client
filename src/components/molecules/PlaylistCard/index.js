import { faPlay } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import React from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";
import { playPlaylist } from "../../../api/playlists";
import GreenPlayButton from "../../atoms/GreenPlayButton";

const Card = styled.div`
  padding: 10px;
  background-color: rgba(33, 33, 33, 1.75);
  width: 200px;
  height: 300px;
  border-radius: 5px;
  color: white;
  display: flex;
  flex-direction: column;
  margin: 20px 10px;
  align-items: center;
`;

const StyledImage = styled.img`
  width: 100%;
  border-radius: 5px;
`;

const StyledTextContainer = styled.div`
  width: 100%;
  padding: 20px 10px;
  text-align: left;
`;

const StyledName = styled.p`
  font-weight: bold;
  margin: 0 0;
`;

const StyledOwner = styled.p`
  color: grey;
  margin: 0;
`;

const StyledPlayButtonOverlay = styled.div`
  height: 40px;
  width: 40px;
  margin: 10px;
  background-color: black;
  opacity: 0;
  border-radius: 45px;
  position: absolute;
  bottom: 0;
  right: 0;
  transition: 0.5s ease;
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  z-index: 1000;

  &:hover {
    opacity: 1;
    box-shadow: 0px 2px 2px 0px black;
  }
`;

const Container = styled.div`
  position: relative;
  &:hover ${StyledPlayButtonOverlay} {
    opacity: 1;
    box-shadow: 0px 2px 2px 0px black;
  }
`;

const PlaylistCard = ({ playlist }) => {
  const imageUrl = playlist.images[0] && playlist.images[0].url;
  const path = `/playlists/${playlist.id}`;

  return (
    <Link to={path} style={{ textDecoration: "none" }}>
      <Card>
        <Container>
          <StyledImage src={imageUrl} />
          <StyledPlayButtonOverlay>
            <GreenPlayButton action={() => playPlaylist(playlist.id)} />
          </StyledPlayButtonOverlay>
        </Container>
        <StyledTextContainer>
          <StyledName>{playlist.name}</StyledName>
          <StyledOwner>By {playlist.owner.display_name}</StyledOwner>
        </StyledTextContainer>
      </Card>
    </Link>
  );
};

export default PlaylistCard;
