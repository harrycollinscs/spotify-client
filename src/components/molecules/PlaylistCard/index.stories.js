import React from "react";
import PlaylistCard from "./index";
import PlaylistData from "../../../__fixtures__/Playlist/index.json";

export default {
  title: "Molecules/PlaylistCard",
  component: PlaylistCard,
};

const Template = (args) => <PlaylistCard {...args} />;

export const Default = Template.bind({});
Default.args = {
  playlist: PlaylistData,
};

