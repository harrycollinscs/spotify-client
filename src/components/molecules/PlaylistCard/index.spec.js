import React from "react";
import PlaylistCard from "./index";
import { shallow } from "enzyme";
import PlaylistData from "../../../__fixtures__/Playlist/index.json"

describe("Molecules/PlaylistCard", () => {
  const props = {
    playlist: PlaylistData,
  };

  const component = <PlaylistCard {...props} />;

  it("Should match snapshot", () => {
    const render = shallow(component);
    expect(render).toMatchSnapshot();
  });
});
