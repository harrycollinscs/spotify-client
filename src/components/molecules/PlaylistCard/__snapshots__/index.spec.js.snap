// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/PlaylistCard Should match snapshot 1`] = `
<Link
  style={
    Object {
      "textDecoration": "none",
    }
  }
  to="/playlists/3qQoF3PaRtg1iPUHKRzjEX"
>
  <styled.div>
    <styled.div>
      <styled.img
        src="https://mosaic.scdn.co/640/ab67616d0000b2732090f4f6cc406e6d3c306733ab67616d0000b27352c75ed37313b889447011efab67616d0000b27364442a4144ef18be67b37b37ab67616d0000b273ea9bbdf9f1433e68c7dffbad"
      />
      <styled.div>
        <GreenPlayButton
          action={[Function]}
        />
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.p>
        ðŸš¿
      </styled.p>
      <styled.p>
        By 
        woodpush3r
      </styled.p>
    </styled.div>
  </styled.div>
</Link>
`;
